src/backend/mod.rs

```diff
@@ -85,4 +89,24 @@ impl DNSBackend {
 
         DNSResult::Success(results)
     }
+
+    // reverse lookup must return a single name resolved via mapping
+    pub fn reverse_lookup(&self, requester: &IpAddr, lookup_ip: &IpAddr) -> Vec<String> {
+        let nets = match self.ip_mappings.get(requester) {
+            Some(n) => n,
+            None => return vec![],
+        };
+
+        for net in nets {
+            match self.reverse_mappings.get(net) {
+                Some(n) => match n.get(lookup_ip) {
```

Comment:
I'd use `if let Some(n) = self.reverse_mappings.get(net) {` if your `None` case is just `{}`.